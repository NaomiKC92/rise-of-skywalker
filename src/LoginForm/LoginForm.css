/* * {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  background-attachment: fixed;
  background-image: url('../images/login-bg.jpg')
} */

