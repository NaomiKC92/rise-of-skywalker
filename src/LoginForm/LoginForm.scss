* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.login-page {
  background-attachment: fixed;
  background-image: url('../images/login-duel.webp');
  background-size: cover;
  background-repeat: no-repeat;
  height: 120vh;
}

h1 {
  color: #fff;
  text-align: center;
  text-shadow: -1px 0 black, 0 2px black, 0px 0 black, 0 -1px black;
  font-size: 50px;
  padding: 35px;
}

@font-face {
  font-family: 'Star-Jedi';
  src: url('../fonts/starjedi-webfont.woff') format('woff');
}

form {
  display: flex;
  flex-direction: column;
  width: 40%;
  align-self: center;
  margin: 3% 0 0 30%;
  justify-content: space-between;
  height: 200px;
}

input {
  height: 40px;
  font-size: 20px;
  border: 2px solid #000;
  padding: 10px;
  border-radius: 4px;
}

::placeholder {
  color: #141d4f;
}

.login-btn,
select {
  cursor: pointer;
  border: 2px solid #000;
  height: 40px;
  font-size: 20px;
  color: #141d4f;
  border-radius: 4px;
}

.link {
  display: block;
  margin: auto;
}

.show-error {
  color: red;
}

.do-not-show-error {
  visibility: hidden;
}
