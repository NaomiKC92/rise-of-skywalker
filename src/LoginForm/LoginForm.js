import React, { Component } from 'react';
import './LoginForm.scss';
import { Link } from 'react-router-dom';

class LoginForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      quote: '',
      rank: 'Youngling'
    };
  }

  resetInputFields = event => {
    this.setState({
      name: '',
      quote: '',
      rank: 'Youngling'
    });
  };

  handleChange = event => {
    event.preventDefault();
    this.setState({ [event.target.name]: event.target.value });
  };

  handleSubmit = () => {};

  render() {
    return (
      <div className='login-page'>
        <h1>Rise of Skywalker</h1>
        <form>
          <input
            type='text'
            name='name'
            value={this.state.name}
            placeholder='Enter Your Name. Ex: Finn'
            onChange={event => this.handleChange(event)}
          />
          <input
            type='text'
            name='quote'
            value={this.state.quote}
            placeholder='Enter Your Favorite Quote. Ex: Finn is the best'
            onChange={event => this.handleChange(event)}
          />
          <select
            name='rank'
            value={this.state.rank}
            onChange={e => this.handleChange(e)}>
            <option value='Youngling'>Youngling</option>
            <option value='Padawan'>Padawan</option>
            <option value='Jedi Knight'>Jedi Knight</option>
            <option value='Jedi Master'>Jedi Master</option>
          </select>
          <Link className='link' to={`/movies`}>
            <button
              type='button'
              className='login-btn'
              onClick={this.resetInputFields}>
              Submit to Proceed
            </button>
          </Link>
        </form>
      </div>
    );
  }
}

export default LoginForm;
