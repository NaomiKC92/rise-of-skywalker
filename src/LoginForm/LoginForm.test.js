import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

describe('LOGIN FORM', () => {
  let wrapper;

  const mockSetupUserProfile = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<LoginForm setupUserProfile={mockSetupUserProfile} />);
  });

  it('should match the LoginForm Component Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update the appropriate states for user profile when handleChange is called', () => {
    wrapper = shallow(<LoginForm setupUserProfile={mockSetupUserProfile} />);

    const mockEvent1 = {
      target: {
        name: 'name',
        value: 'Peerat'
      }
    };

    const mockEvent2 = {
      target: {
        name: 'quote',
        value:
          "If there ever comes a day when we can't be together, keep me in your heart. I'll stay there forever."
      }
    };

    const mockEvent3 = {
      target: {
        name: 'rank',
        value: 'Jedi Knight'
      }
    };

    wrapper.instance().handleChange(mockEvent1);
    expect(wrapper.state('name')).toEqual('Peerat');

    wrapper.instance().handleChange(mockEvent2);
    expect(wrapper.state('quote')).toEqual(
      "If there ever comes a day when we can't be together, keep me in your heart. I'll stay there forever."
    );

    wrapper.instance().handleChange(mockEvent3);
    expect(wrapper.state('rank')).toEqual('Jedi Knight');
  });

  it('should call on setupUserProfile when submit to login to movies page is clicked (handleSubmit)', () => {
    wrapper = shallow(<LoginForm setupUserProfile={mockSetupUserProfile} />);
    wrapper.setState({
      name: 'Peerat',
      quote: 'Winnie the Pooh!',
      rank: 'Jedi Knight'
    });
    wrapper
      .find('button')
      .at(0)
      .simulate('click');

    expect(mockSetupUserProfile).toHaveBeenCalled();
  });
});
