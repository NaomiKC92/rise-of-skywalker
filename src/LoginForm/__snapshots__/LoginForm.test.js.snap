// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOGIN FORM should match the LoginForm Component Snapshot 1`] = `
<div
  className="login-page"
>
  <h1>
    Rise of Skywalker
  </h1>
  <form>
    <h2
      className="do-not-show-error"
    >
      Please make sure to have all the information to get started!
    </h2>
    <input
      name="name"
      onChange={[Function]}
      placeholder="Enter Your Name. Ex: Finn"
      type="text"
      value=""
    />
    <input
      name="quote"
      onChange={[Function]}
      placeholder="Enter Your Favorite Quote. Ex: Finn is the best"
      type="text"
      value=""
    />
    <select
      name="rank"
      onChange={[Function]}
      value="Youngling"
    >
      <option
        value="Youngling"
      >
        Youngling
      </option>
      <option
        value="Padawan"
      >
        Padawan
      </option>
      <option
        value="Jedi Knight"
      >
        Jedi Knight
      </option>
      <option
        value="Jedi Master"
      >
        Jedi Master
      </option>
    </select>
    <button
      className="login-btn"
      onClick={[Function]}
      type="button"
    >
      Submit to Proceed
    </button>
  </form>
</div>
`;
